### PR TITLE
Add documentation on the oracle session environment

### LINKED ISSUE #217: OracleSessionInit is NOT loaded properly in the Dependency Injection
Hi,
I moved my project from MySQL to Oracle, and I get this error when I try to insert into the database.

An exception occurred while executing 'INSERT INTO arborescence (IDNT_ARBR, NOM_ARBR, DATE_CREATION, IDNT_SERV, IDNT_UTLS) VALUES (?, ?, ?, ?, ?)' with params [7, "B", "2013-10-09 00:00:00", 7, 1]:
ORA-01861: literal does not match format string

I solved this problem by adding the following code to : app/config/config.yml

services:
    oracle.listener:
        class: Doctrine\DBAL\Event\Listeners\OracleSessionInit 
        tags:
            - { name: doctrine.event_listener, event: postConnect }

This seems to be a bug in the DoctrineBundle. If we use the MySQL driver, the corresponding MysqlSessionInit is loaded properly in the Dependency Injection extension. But this doesn't happen with the OracleSessionInit class, if you we the Oracle driver.

Respectfully,
iftah

### PR DESCRIPTION
**Pull request by mikeSimonson**
Fix #217 and replace #574

### CODE REVIEW

**Comment by mikeSimonson** [comment]
@kimhemsoe Can you validate that explanation ?

File: Resources/doc/configuration.rst [hunk 1]
```diff
@@ -722,6 +722,32 @@ the ORM resolves to:
 There are lots of other configuration options that you can use to overwrite
 certain classes, but those are for very advanced use-cases only.
 
+Oracle
+=======
+
+If the environment format configured in oracle does not meet doctrine requirements,
+you need to use OracleSessionInit listener so that doctrine is aware of the format used by oracle.
```
Comment (Ocramius) [inline Resources/doc/configuration.rst]: `use the`

Comment (Ocramius) [inline Resources/doc/configuration.rst]: `by Oracle DB`

File: Resources/doc/configuration.rst [hunk 2]
```diff
@@ -722,6 +722,32 @@ the ORM resolves to:
 There are lots of other configuration options that you can use to overwrite
 certain classes, but those are for very advanced use-cases only.
 
+Oracle
```
Comment (Ocramius) [inline Resources/doc/configuration.rst]: `Oracle DB`

File: Resources/doc/configuration.rst [hunk 3]
```diff
@@ -722,6 +722,32 @@ the ORM resolves to:
...
+
+.. code-block:: yaml
+
+    services:
+        oracle.listener:
+            class: Doctrine\DBAL\Event\Listeners\OracleSessionInit
+            tags:
+                - { name: doctrine.event_listener, event: postConnect }
+
+The environment variables needed for doctrine and their default format are:
```
Comment (Ocramius) [inline Resources/doc/configuration.rst]: These are environment vars for Oracle DB, not Doctrine DBAL: the [listener](https://github.com/doctrine/dbal/blob/b551fcb26a81e230085c0dca88174573b6c36609/lib/Doctrine/DBAL/Event/Listeners/OracleSessionInit.php) doesn't really use the environment, AFAIK.

**Comment by mikeSimonson** [comment]
@Ocramius Thanks

**Comment by Ocramius** [comment]
LGTM! 👍

**Pull request merged**